File: TMTDentalAPI/ClientApp/src/app/account-invoices/account-invoice-line-display.ts
import { ToothDisplay } from '../teeth/tooth.service';
import { ToothCategoryBasic } from '../tooth-categories/tooth-category.service';
import { ProductSimple } from '../products/product-simple';
import { PartnerSimple } from '../partners/partner-simple';

export class AccountInvoiceLineDisplay {
    id: string;
    name: string;
    uomId: string;
    product: ProductSimple;
    productId: string;
    employee: PartnerSimple;
    employeeId: string;
    accountId: string;
    priceUnit: number;
    priceSubTotal: number;
    discount: number;
    quantity: number;
    teeth: ToothDisplay[];
    toothCategory: ToothCategoryBasic;
}


export class AccountInvoiceLinePaged {
    offset: number;
    limit: number;
    invoiceId: string;
    productId: string;
    accountId: string;
    partnerId: string;
    toothCategoryId: string;
    diagnostic: string;
}

File: Umbraco.Web/Models/ContentEditing/AccountInvoiceLineDisplay.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Umbraco.Web.Models.ContentEditing
{
    public class AccountInvoiceLineDisplay
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public int Sequence { get; set; }

        public Guid? UoMId { get; set; }

        public Guid? ProductId { get; set; }
        public ProductSimple Product { get; set; }

        public Guid AccountId { get; set; }

        public decimal PriceUnit { get; set; }
        public decimal PriceSubTotal { get; set; }
        public decimal Quantity { get; set; }

        /// <summary>
        /// Chuẩn đoán
        /// </summary>
        public string Diagnostic { get; set; }

        public Guid? ToothCategoryId { get; set; }
        public ToothCategoryBasic ToothCategory { get; set; }

        public Guid? ToothId { get; set; }
        public ToothBasic Tooth { get; set; }

        public Guid? InvoiceId { get; set; }
        public AccountInvoiceSimple Invoice { get; set; }

        public IEnumerable<ToothDisplay> Teeth { get; set; } = new List<ToothDisplay>();
    }
}
